addons/account/models/account_move_line.py

```diff
@@ -1108,13 +1023,11 @@ def _compute_epd_needed(self):
                     'name': _("Early Payment Discount (%s)", discount_percentage_name),
                     'amount_currency': 0.0,
                     'balance': 0.0,
-                    'price_subtotal': 0.0,
                     'tax_ids': [Command.clear()],
                 },
             )
-            epd_needed_vals['amount_currency'] += line.currency_id.round(line.amount_currency * percentage)
-            epd_needed_vals['balance'] += line.currency_id.round(line.balance * percentage)
-            epd_needed_vals['price_subtotal'] += line.currency_id.round(line.price_subtotal * percentage)
+            epd_needed_vals['amount_currency'] += amount_currency
+            epd_needed_vals['balance'] += balance
```

Comment:
what a mess... I wouldn't mind a small refactoring that simply sets the `epd_needed_vals` without the fancy setDefault, frozenDict and default value dicts... :facepalm:

Reply:
You mean like what we had before v16 with a single method per dynamic type that was creating directly the account.move.line without a fancy sync engine behind nobody understand? :roll_eyes:

Reply:
#TODO added for post OXP